### PR TITLE
Fix for issue 1499: UIView: [UIView invalidateIntrinsicContentSize] does far more work than necessary

### PR DESCRIPTION
**Pull request by jaredhms**
Moving to a lazy model for constraint recalculation allows IslandwoodNews to launch more reliably on phone, but additional perf investigation is ongoing.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/1506)

### CODE REVIEW

File: Frameworks/AutoLayout/AutoLayout.mm [hunk 1]
```diff
@@ -517,6 +496,58 @@ - (void)autoLayoutInvalidateContentSize {
...
 }
 
+- (void)autoLayoutLayoutSubviews {
+    UIView* topView = [self autolayoutRoot];
+
+    // Make sure our constraints are up to date
+    [self _validateContentSize];
+    [topView _validateContentSize];
+
+    for (int i = 0; i < [self.subviews count]; i++) {
```
Comment (aballway) [inline Frameworks/AutoLayout/AutoLayout.mm]: nit: for in loop

File: Frameworks/UIKit/StarboardXaml/ApplicationCompositor.cpp [hunk 2]
```diff
@@ -59,19 +59,24 @@ extern "C" void RunApplicationMain(Platform::String^ principalClassName,
                                    float windowHeight,
                                    ActivationType activationType,
                                    Platform::Object^ activationArg) {
+    // Return from activation immediatly; push all subsequent app launch work to
```
Comment (aballway) [inline Frameworks/UIKit/StarboardXaml/ApplicationCompositor.cpp]: This seems unrelated to AutoLayout issues, and should be tested really thoroughly to make sure we don't accidentally regress anywhere with app launching

File: Frameworks/AutoLayout/AutoLayout.mm (hunk 1 above)
Comment (jaredhms) [inline Frameworks/AutoLayout/AutoLayout.mm]: going to leave as-is for now; not new code.

File: Frameworks/UIKit/StarboardXaml/ApplicationCompositor.cpp (hunk 2 above)
Comment (jaredhms) [inline Frameworks/UIKit/StarboardXaml/ApplicationCompositor.cpp]: Agreed; it's an additional 'fix' for the app launch timeouts.  Raj had intended to do this with a previous change, but it didn't push all work off of the initial activation.  I agree that we need to test adequately.

**Comment by yiyang-msft** [comment]
:shipit:

**Comment by jaredhms** [comment]
I have a much larger change for this; we'll roll it out in a separate PR.
